src/jdk.incubator.concurrent/share/classes/jdk/incubator/concurrent/ScopedValue.java

```diff
@@ -220,10 +253,30 @@ static final Snapshot getInstance() {
     }
 
     /**
-     * An immutable map of scoped values to values.
-     * It define the {@link #run(Runnable) run} and {@link #call(Callable) call} methods
-     * to invoke an operation with the scoped value mappings bound to the thread
-     * that invokes {@code run} or {@code call}.
+     * A mapping of scoped values, as <em>keys</em>, to values.
+     *
+     * <p> A {@code Carrier} is used to accumlate mappings so that an operation (a
+     * {@link Runnable} or {@link Callable}) can be executed with all scoped values in
+     * the mapping bound to a value. The following example runs an operation with k1
+     * bound (or rebound) to v1, and k2 bound (or rebound) to v2.
+     * {@snippet lang=java :
+     *     // @link substring="where" target="#where(ScopedValue, Object)" :
+     *     ScopedValue.where(k1, v1).where(k2, v2).run(() -> ... );
+     * }
+     * It is equivalent to:
+     * {@snippet lang=java :
+     *     // @link substring="where" target="ScopedValue#where(ScopedValue, Object, Runnable)" :
+     *     ScopedValue.where(k1, v1, () -> {
+     *         ScopedValue.where(k2, v2, () -> ... );
+     *     });
+     * }
+     *
```

Comment:
This is a bit circular: we define
```
ScopedValue.where(key, value, op)
``` 
as equivalent to
```
ScopedValue.where(key, value).run(op);
```
then we define 
```
ScopedValue.where(k1, v1).where(k2, v2).run(op);
```
as equivalent to
```
ScopedValue.where(k1, v1, () -> {
        ScopedValue.where(k2, v2, () -> ... );
    });
```

so we're defining them in terms of each other. I guess that's OK.

Reply:
The intention here was to show that the chaining avoids needing to nest the lambda expressions. I'll make to make this better to avoid the circularity.